### PR TITLE
Add object notation support for `initializeCommand`

### PR DESCRIPTION
**Pull request by joshspicer**
refs: https://github.com/microsoft/vscode-remote-release/issues/7765, https://github.com/devcontainers/spec/issues/168, https://github.slack.com/archives/C0431R35H37/p1683554998695899

Aligns the `initializeCommand` lifecycle hook with the other hooks, allow for the 'object notation', running each provided (and named) command in parallel with one another.

For logging, the behavior is mirrored for the other lifecycle hooks, _not_ continuously streaming the output from each parallel hook, but instead holding the output until the end (otherwise, output will be interleaved and hard to follow).  See the demo below for an illustration of this.

https://user-images.githubusercontent.com/23246594/236972162-860377bb-0337-487b-9dd9-67440ebacefb.mov

### CODE REVIEW

**Review by chrmarti — APPROVED** [review]
LGTM!

**Comment by justin-mccann** [comment]
Will this PR also fixup VSCode's problem-highlighting on the acceptable types for `initializeCommand`, or are there separate changes necessary for that?

**Comment by joshspicer** [comment]
> Will this PR also fixup VSCode's problem-highlighting on the acceptable types for `initializeCommand`, or are there separate changes necessary for that?

Separate changes are needed for that. Specifically, an update to the schema published here: https://raw.githubusercontent.com/devcontainers/spec/main/schemas/devContainer.base.schema.json

**Comment by joshspicer** [comment]
Manually confirming things look good with the existing syntaxes

image
image
